src/components/context-selection/PeriodSelector.js

```diff
@@ -1,20 +1,195 @@
-import { SelectorBarItem } from '@dhis2/ui'
+import i18n from '@dhis2/d2-i18n'
+import { generateFixedPeriods } from '@dhis2/multi-calendar-dates'
+import {
+    Button,
+    InputField,
+    SelectorBarItem,
+    SingleSelectField,
+    SingleSelectOption,
+} from '@dhis2/ui'
 import PropTypes from 'prop-types'
 import React, { useState } from 'react'
+import styles from './PeriodSelector.module.css'
 
-export const PeriodSelector = ({ selectedPeriodInfo }) => {
+const ANNUAL_REPORT_PERIOD_TYPES = [
+    { label: i18n.t('Yearly'), id: 'YEARLY' },
+    { label: i18n.t('Financial Year (April)'), id: 'FYAPR' },
+    { label: i18n.t('Financial Year (July)'), id: 'FYJUL' },
+    { label: i18n.t('Financial Year (October)'), id: 'FYOCT' },
+    // todo: wait, what? should this be Nov? see /api/periodTypes
+    { label: i18n.t('Financial Year (December)'), id: 'FYDEC' },
+]
+
+function PeriodTypeSelect({ periodType, setPeriodType, setPeriods }) {
+    return (
+        <SingleSelectField
+            label={i18n.t('Period type')}
+            placeholder={i18n.t('Choose a period type')}
+            selected={periodType}
+            onChange={({ selected }) => {
+                setPeriodType(selected)
+                // Reset period selection too
+                setPeriods([])
+            }}
+        >
+            {ANNUAL_REPORT_PERIOD_TYPES.map(({ label, id }) => (
+                <SingleSelectOption key={id} label={label} value={id} />
+            ))}
+        </SingleSelectField>
+    )
+}
+PeriodTypeSelect.propTypes = {
+    /** Expected to be an ISO ID of the period type */
+    periodType: PropTypes.string,
+    setPeriodType: PropTypes.func,
+    setPeriods: PropTypes.func,
+}
+
+/** Combines period and yearsForReference because their state is shared */
+const PeriodsSelect = ({
+    periodType,
+    periods,
+    setPeriods,
+    yearsForReference,
+    setYearsForReference,
+}) => {
+    const generatedPeriods = React.useMemo(() => {
+        return periodType
+            ? generateFixedPeriods({
+                  // todo: should this be systemInfo.calendar?
+                  calendar: 'gregory',
+                  periodType: periodType,
+                  year: new Date().getFullYear() - 1,
+                  // todo: internationalize; me.settings.keyUiLocale (or DB locale?)
+                  locale: 'en',
+              })
+            : []
+    }, [periodType])
+
+    const handleChange = React.useCallback(
+        ({ periodId, newYearsForReference }) => {
+            if (!periodId) {
+                return
+            }
+
+            const selectedPeriodIdx = generatedPeriods.findIndex(
+                ({ id }) => id === periodId
+            )
+            const endIdx = selectedPeriodIdx + newYearsForReference + 1
+            // todo: add an alert to show warning in UI. Could also validate years > 0
+            if (endIdx > generatedPeriods.length) {
```

Comment:
We can actually generate more periods beyond the 10 years, you just have to rerun generateFixedPeriods with `year: new Date().getFullYear() -11`

But...then I checked data entry app, and we are just passing yearsCount to `generateFixedPeriod`, but we're also using the alpha release of multi-calendar-dates. 

Maybe we can leave as is and follow up with Mozafar? I feel like we might as well let people select back to at least 1990 to be with the old app.

I think if we select a date far enough back, it's fine to have a console.warn, but with a period of only 10 years, this might be something we'd want to alert users about?

Reply:
We can also end up with 1 period with the current check, e.g. when the last of the ten generated periods is selected and then we have no comparison periods to select.

But, I think it's unlikely (particularly if we have more periods), and we will at some point need to implement handling for no comparison periods when making calculations.